jerry-main/main-unix.c

```diff
@@ -407,7 +407,7 @@ check_usage (bool condition, /**< the condition that must hold */
   }
 } /* check_usage */
 
-#ifdef JERRY_ENABLE_EXTERNAL_CONTEXT
+#if defined (JERRY_EXTERNAL_CONTEXT) && (JERRY_EXTERNAL_CONTEXT == 1)
```

Comment:
Wouldn't be a better idea, to use the `ENABLED (JERRY_EXTERNAL_CONTEXT)` macro here? (Same as all occurrences).

Reply:
Would be good idea, but the `config.h` where the macro is defined is not accessible outside of jerry-core. In addition even using these defines are questionable a bit as they are not correctly "exported" via headers.

Reply:
I think exporting the `ENABLED/DISABLED` macros out of the `jerry-core`'s scope for only this guard in unnecessary. But as you mentioned the `JERRY_DEBUGGER` macro also crosses layers (core/ext/port) and for the consistency IMO it would be great to use the `ENABLED/DISABLED` as well which also requires the export these macros to headers. So, I think propagating these macros would be useful. What is your opinion about it?

Reply:
Yes, I would also like to expose the macros outside of jerry-core. I have an idea for it, but it should be another PR if it is ok for you.

Reply:
Sure!